### PR TITLE
version '1.3.28': StoredValue

### PR DESCRIPTION
**Pull request by van-agustin**

### CODE REVIEW

File: src/main/java/au/com/dmg/fusion/request/storedvaluerequest/StoredValueAccountStatus.java
Comment (i-grace) [inline src/main/java/au/com/dmg/fusion/request/storedvaluerequest/StoredValueAccountStatus.java:1]: StoredValueAccountStatus is part of the response - not the request.

File: src/main/java/au/com/dmg/fusion/request/storedvaluerequest/StoredValueData.java
Comment (i-grace) [inline src/main/java/au/com/dmg/fusion/request/storedvaluerequest/StoredValueData.java:1]: EanUpc is a required field.

File: src/main/java/au/com/dmg/fusion/response/storevalueresponse/StoredValueResponse.java
Comment (i-grace) [inline src/main/java/au/com/dmg/fusion/response/storevalueresponse/StoredValueResponse.java:1]: If response.Result = Success, storedValueResult must not be null and should contain at least 1 item.

File: src/main/java/au/com/dmg/fusion/response/storevalueresponse/StoredValueResult.java
Comment (i-grace) [inline src/main/java/au/com/dmg/fusion/response/storevalueresponse/StoredValueResult.java:1]: should also include StoredValueAccountStatus.

File: src/test/java/au/com/dmg/fusion/response/StoredValueResponseTest.java
Comment (i-grace) [inline src/test/java/au/com/dmg/fusion/response/StoredValueResponseTest.java:1]: Please add test cases for when: 
- result is Failure and StoredValueResult is missing, no exception is thrown. 
- result is Success and StoredValueResult is missing, exception is thrown.
- result is Success and StoredValueResult is present, no exception is thrown.

File: src/test/java/au/com/dmg/fusion/request/StoredValueRequestTest.java
Comment (i-grace) [inline src/test/java/au/com/dmg/fusion/request/StoredValueRequestTest.java:1]: Please add test case for when EanUPC is missing.

**Review by i-grace — CHANGES_REQUESTED** [review]

File: src/main/java/au/com/dmg/fusion/response/storevalueresponse/StoredValueResult.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            this.storedValueAccountStatus = storedValueAccountStatus;
+        }
+
+        public Builder storedValueAccountStatus(StoredValueAccountStatus storedValueAccountStatus){
+            this.storedValueAccountStatus = storedValueAccountStatus;
+            return Builder.this;
+        }
+
+        public StoredValueResult build() {
+            if (this.storedValueTransactionType == null) {
```
Comment (i-grace) [inline src/main/java/au/com/dmg/fusion/response/storevalueresponse/StoredValueResult.java:29]: IdentificationType and StoredValueID are also required.

**Review by i-grace — CHANGES_REQUESTED** [review]

File: src/main/java/au/com/dmg/fusion/response/storevalueresponse/StoredValueResult.java (hunk 1 above)
Comment (van-agustin) [reply src/main/java/au/com/dmg/fusion/response/storevalueresponse/StoredValueResult.java:29]: Hi @i-grace , the identificationType and storedValueID are required under StoredValueAccountStatus. I updated the test cases instead to capture this.

Comment (i-grace) [reply src/main/java/au/com/dmg/fusion/response/storevalueresponse/StoredValueResult.java:29]: There's currently validation in StoredValueAccountStatus for those fields?

public StoredValueAccountStatus build(){
            return new StoredValueAccountStatus(this);
        }

**Review by i-grace — CHANGES_REQUESTED** [review]

File: src/main/java/au/com/dmg/fusion/response/storevalueresponse/StoredValueResult.java (hunk 1 above)
Comment (van-agustin) [reply src/main/java/au/com/dmg/fusion/response/storevalueresponse/StoredValueResult.java:29]: Sorry, it's under StoredValueAccountID @i-grace

Comment (van-agustin) [reply src/main/java/au/com/dmg/fusion/response/storevalueresponse/StoredValueResult.java:29]: Hi @i-grace -- I just added the null checking on the StoredValueAccountStatus build.

**Pull request merged**